drivers/hmis/app/models/hmis/hud/validators/enrollment_validator.rb

```diff
@@ -34,17 +34,25 @@ def self.already_enrolled_full_message
   end
 
   def self.validate_entry_date(enrollment, household_members: nil, options: {})
-    entry_date = enrollment.entry_date
+    entry_date = enrollment&.entry_date
     return [] unless entry_date.present?
 
     errors = HmisErrors::Errors.new
     dob = enrollment.client&.dob
-    exit_date = enrollment&.exit_date
+    exit_date = enrollment.exit_date
 
     errors.add :entry_date, :out_of_range, message: future_message, **options if entry_date.future?
     errors.add :entry_date, :out_of_range, message: over_twenty_years_ago_message, **options if entry_date < (Date.today - 20.years)
     errors.add :entry_date, :out_of_range, message: before_dob_message, **options if dob.present? && dob > entry_date
     errors.add :entry_date, :out_of_range, message: after_exit_message(exit_date), **options if exit_date.present? && exit_date < entry_date
+
+    conflict_scope = enrollment.project
+      .enrollments_including_wip
+      .where(personal_id: enrollment.personal_id, data_source_id: enrollment.data_source_id)
+      .with_conflicting_dates(enrollment.entry_date...)
+
+    conflict_scope = conflict_scope.where.not(id: enrollment.id) if enrollment.persisted?
+    errors.add(:entry_date, :out_of_range, full_message: already_enrolled_full_message) if conflict_scope.any?
```

Comment:
On second thought. Maybe we also make this a warning, unless it's a brand new enrollment. We should not block the user from doing things they need to do. @eanders ?

```suggestion
    errors.add(:entry_date, :out_of_range, severity: enrollment.new_record? ? :error : :warning, full_message: already_enrolled_full_message) if conflict_scope.any?
```

Reply:
Actually to accommodate @eanders's concern with https://github.com/greenriver/hmis-warehouse/pull/3428/files#r1307365655, this would need to be a warning even for new enrollment creation.

Reply:
@ttoomey I just synced up with Elliot about what the behavior here should be. We landed on:
* If the entry date is being changed on an EXISTING enrollment, and it overlaps with another one, it should be a **warning**
* If the entry date is being set on a NEW enrollment, and the entry date is _on or after_ the entry date of any conflicting enrollments, it should be an **error**.
* If the entry date is being set on a NEW enrollment, and the entry date is _before_ the entry date of any conflicting enrollments, it should be a **warning**. This allows for after-the-fact data entry. For example someone is currently enrolled at the program, but you need to add in an enrollment that they had last month.

Since you have the `conflict_scope` available here, I think that should be doable. Let me know if its not clear.!

Reply:
Thanks for the clear summary! I believe this is done